draft-ietf-dprive-dnsoquic.md

```diff
@@ -375,6 +381,8 @@ messages during a transaction. These include (but are not limited to)
 * a client receives a different number of responses on a stream than expected
   (e.g. multiple responses to a query for an A record)
 * a client receives a STOP_SENDING request
+* the maximum number of dangling streams exceeds the limits set by the implementation
+  (see {{stream-mapping-and-usage}})
```

Comment:
I'm not sure this is the best example - is this really a protocol error to exceed an arbitrary limit set by the server that a client can't know? All our other examples here are specific one-off events that show a peer is clearly non-compliant. 

This does feel tricky. Our current text actually makes no specification for *when* the FIN must be sent in relation to the query, so we probably need to tighten that before we can declare this a fatal protocol error. Could we say something like:
`The client MUST send the DNS query over the selected stream, and MUST indicate immediately
through the STREAM FIN mechanism that no further data will be sent on that
stream.`
and then declare a dangling stream if the server sends a FIN before receiving one? In which case exceeding a limit of dangling connections feels closer to the strategy of limiting transaction cancellation with DOQ-EXCESSIVE_LOAD than a protocol error.

Reply:
QUIC server stacks typically maintain a count of open streams, and only give credits for new ones when that count is below a threshold. If the client delays sending FIN, the server will typically delay sending new credits for new streams, and the client will not be able to start new transactions.

Reply:
@saradickinson, would it be better if the text was updated to say "* the client or server does not indicate the expected STREAM FIN after sending requests or responses." That would focus the definition of the error on the bad behavior, instead of on the way by which it is detected.